src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -0,0 +1,358 @@
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(left, right);
+
+        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
+
+        /// <inheritdoc cref="EqualsIgnoreCase(ReadOnlySpan{byte}, ReadOnlySpan{byte})"/>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
+            => left.Length == right.Length && SequenceEqualIgnoreCase(right, left);
+
+        private static bool SequenceEqualIgnoreCase<TLeft, TRight>(ReadOnlySpan<TLeft> left, ReadOnlySpan<TRight> right)
+            where TLeft : unmanaged, INumberBase<TLeft>
+            where TRight : unmanaged, INumberBase<TRight>
+        {
+            Debug.Assert(left.Length == right.Length);
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                uint valueA = uint.CreateTruncating(left[i]);
+                uint valueB = uint.CreateTruncating(right[i]);
+
+                if (!UnicodeUtility.IsAsciiCodePoint(valueA) || !UnicodeUtility.IsAsciiCodePoint(valueB))
+                {
+                    return false;
+                }
+
+                if (valueA == valueB)
+                {
+                    continue; // exact match
+                }
+
+                valueA |= 0x20u;
+                if ((uint)(valueA - 'a') > (uint)('z' - 'a'))
+                {
+                    return false; // not exact match, and first input isn't in [A-Za-z]
+                }
+
+                if (valueA != (valueB | 0x20u))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<ushort> Widen(Vector64<byte> bytes)
+        {
+            if (AdvSimd.IsSupported)
+            {
+                return AdvSimd.ZeroExtendWideningLower(bytes);
+            }
+            else
+            {
+                (Vector64<ushort> lower, Vector64<ushort> upper) = Vector64.Widen(bytes);
+                return Vector128.Create(lower, upper);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector256<ushort> Widen(Vector128<byte> bytes)
+        {
+            (Vector128<ushort> lower, Vector128<ushort> upper) = Vector128.Widen(bytes);
+            return Vector256.Create(lower, upper);
+        }
+
+        private static bool VectorContainsNonAsciiChar(Vector64<byte> bytes)
+            => !Utf8Utility.AllBytesInUInt64AreAscii(bytes.AsUInt64().ToScalar());
```

Comment:
Same concern here for Vector64. Pass in `ulong` as `Utf8Utility.AllBytesInUInt64AreAscii` expects that.
So get rid of Vector64 where possible and use `ulong` instead?